zkvm/src/types.rs

```diff
@@ -63,6 +64,7 @@ pub struct Variable {
 pub struct Expression {
     /// Terms of the expression
     pub(crate) terms: Vec<(r1cs::Variable, Scalar)>,
+    pub(crate) assignment: Option<SignedInteger>,
```

Comment:
Probably we need to allow Scalar assignments too - in case constraints are done over the whole group order |G|, not just integers. 

```suggestion
    pub(crate) assignment: Option<ScalarKind>,
```